src/battle-animations.ts

```diff
@@ -669,7 +669,7 @@ class BattleScene {
 				} else {
 					pokemonhtml += `<span${tooltipCode} style="` + Dex.getPokemonIcon('pokeball') + `" aria-label="Non-statused"></span>`;
 				}
-			} else if (iconType === 'pseudo-zoroark') {
+			} else if (iconType === 'pseudo-zoroark' && this.gen < 8) {
```

Comment:
In addition to being an incorrect interpretation of mechanics, just adding this condition will make the `gen >= 8` condition fall down to `if (!poke)`, turning a flag into an extra pokeball which is clearly not your intent.